src/Functions/castOrDefault.cpp

```diff
@@ -376,66 +376,475 @@ SELECT accurateCastOrDefault('abc', 'UInt32')
+
     factory.registerFunction("toDecimal64OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal64OrDefault", createDecimalMaxPrecision<Decimal64>(0)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal64OrDefault", createDecimalMaxPrecision<Decimal64>(0)); },
+        toDecimal64OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal128OrDefault_description = R"(
+Like [`toDecimal128`](#todecimal128), this function converts an input value to a value of type [Decimal(38, S)](../data-types/decimal.md) but returns the default value in case of an error.
+    )";
+    FunctionDocumentation::Syntax toDecimal128OrDefault_syntax = "toDecimal128OrDefault(expr, S[, default])";
+    FunctionDocumentation::Arguments toDecimal128OrDefault_arguments = {
+        {"expr", "A String representation of a number.", {"String"}},
+        {"S", "Scale parameter between 0 and 38, specifying how many digits the fractional part of a number can have.", {"UInt8"}},
+        {"default", "Optional. The default value to return if parsing to type Decimal128(S) is unsuccessful.", {"Decimal128(S)"}}
+    };
+    FunctionDocumentation::ReturnedValue toDecimal128OrDefault_returned_value = {"Value of type Decimal(38, S) if successful, otherwise returns the default value if passed or 0 if not.", {"Decimal128(S)"}};
+    FunctionDocumentation::Examples toDecimal128OrDefault_examples = {
+        {"Successful conversion", "SELECT toDecimal128OrDefault(toString(1/42), 18)", "0.023809523809523808"},
+        {"Failed conversion", "SELECT toDecimal128OrDefault('Inf', 0, CAST('-1', 'Decimal128(0)'))", "-1"}
+    };
+    FunctionDocumentation::Category toDecimal128OrDefault_category = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation::IntroducedIn toDecimal128OrDefault_introduced_in = {21, 11};
+    FunctionDocumentation toDecimal128OrDefault_documentation = {toDecimal128OrDefault_description, toDecimal128OrDefault_syntax, toDecimal128OrDefault_arguments, toDecimal128OrDefault_returned_value, toDecimal128OrDefault_examples, toDecimal128OrDefault_introduced_in, toDecimal128OrDefault_category};
+
     factory.registerFunction("toDecimal128OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal128OrDefault", createDecimalMaxPrecision<Decimal128>(0)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal128OrDefault", createDecimalMaxPrecision<Decimal128>(0)); },
+        toDecimal128OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal256OrDefault_description = R"(
+Like [`toDecimal256`](#todecimal256), this function converts an input value to a value of type [Decimal(76, S)](../data-types/decimal.md) but returns the default value in case of an error.
```

Comment:
```suggestion
Like [`toDecimal256`](#toDecimal256), this function converts an input value to a value of type [Decimal(76, S)](../data-types/decimal.md) but returns the default value in case of an error.
```